### ISSUE TITLE
Add other browsers

### ISSUE THREAD

**Issue by marco-c** [marco-c/autowebcompat#175]
The more browsers we have, the more screenshots we can collect

**Comment by Shashi456** [comment]
@marco-c i had a couple of doubts regarding this. Is this the same as adding a webdriver for a particular browser and then updating collect.py so that this driver is also run alongside chrome and firefox ?


Apart from that , after we do add a browser won't we have to update data_inconsitencies.py and some functions in utils.py to accommodate these new set of images?

**Comment by marco-c** [comment]
Yes, we will have to add a new webdriver and use it in collect.py and update all the places where we were just expecting Firefox and Chrome.

**Comment by Shashi456** [comment]
So should we add a task list in this Issue ?

**Comment by Shashi456** [comment]
@marco-c having taken a look into it, I've found that safari and edge both are dependent on their respective os and cannot be run cross-platform. 
Should we still pursue adding browsers like opera? or should we add safari and edge driver irrespective of the there not being cross-platform support.

**Comment by marco-c** [comment]
> @marco-c having taken a look into it, I've found that safari and edge both are dependent on their respective os and cannot be run cross-platform.
Should we still pursue adding browsers like opera? or should we add safari and edge driver irrespective of the there not being cross-platform support.

Opera uses Chromium, so there will be no difference between Opera and Chrome.
Yes, I think we should try to add Safari and Edge when we are running the crawler on a supported OS.

**Comment by Vedarth** [comment]
safari is available for windows. So windows supports edge and safari both officially. We can also use wine if we want to use it in linux imo.

**Comment by Shashi456** [comment]
I've tried to run Safari through selenium on Windows , they use an extension which is only compatible with Safari installed on a MacOs.

**Comment by Vedarth** [comment]
Although selenium supports safaridriver in windows, Apple has dropped support :( so It might not work properly. Sorry for the confusion. I think checking the os and choosing edge or safari based on that should be the way to go.

**Comment by marco-c** [comment]
Probably not so valuable to add Edge after all: https://blog.mozilla.org/blog/2018/12/06/goodbye-edge/ :cry: 

But maybe we can use it anyway so we have more diverse screenshots and we're more likely to have incompatibilities in our training set.
